ckanext/canada/view.py

```diff
@@ -579,6 +580,72 @@ def type_redirect(resource_name, dataset_id=None):
                                    owner_org=owner_org_name))
 
 
+@canada_views.route('/recombinant/delete-selected-records/<resource_id>', methods=['GET', 'POST'])
+def delete_selected_records(resource_id):
+    lc = LocalCKAN(username=g.user)
+
+    if request.method != 'POST' or not h.check_access('datastore_records_delete',
+                          {'resource_id': resource_id, 'filters': {}}):
+        abort(403, _('User {0} not authorized to update resource {1}'
+                     .format(str(g.user), resource_id)))
+
+    try:
+        res = lc.action.resource_show(id=resource_id)
+        pkg = lc.action.package_show(id=res['package_id'])
+        org = lc.action.organization_show(id=pkg['owner_org'])
+        dataset = lc.action.recombinant_show(
+            dataset_type=pkg['type'], owner_org=org['name'])
+    except NotFound:
```

Comment:
We should also catch `NotAuthorized` here as well just in case so we can avoid any 500 errors. CKAN and Ian tend to like to return 404's for NotAuthorized errors instead of 403s. So you can just do `except (NotFound, NotAuthorizaed)`